c7n/resources/rds.py

```diff
@@ -2182,3 +2182,34 @@ def process(self, resources, event=None):
                     break
 
         return results
+
+
+@filters.register('has-pending-maintenance')
+class HasPendingMaintenance(Filter):
+    """ Scan DB instances for those with pending maintenance
+    :example:
+    .. code-block:: yaml
+            policies:
+              - name: rds-pending-maintenance
+                resource: rds
+                filters:
+                  - has-pending-maintenance
+    """
+
+    schema = type_schema('has-pending-maintenance')
+    permissions = ('rds:DescribePendingMaintenanceActions',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('rds')
+
+        results = []
+        response = client.describe_pending_maintenance_actions()
+        rds_pending_maintenance_list = [x['ResourceIdentifier']
+                                        for x in response.get('PendingMaintenanceActions')]
+
+        for r in resources:
+            for rds in rds_pending_maintenance_list:
+                if r['DBInstanceArn'] == rds:
+                    results.append(r)
```

Comment:
If we have a set of db identifiers that are pending maintenance, we can avoid a nested loop here:

```suggestion
            if r['DBInstanceArn'] in pending_maintenance:
                results.append(r)
```